2. 유럽국가들에 대해서는 맥주와 와인간의 상관관계는 얼마나 되는지 피어슨상관값(Pearson Correlation)을 구하기
(유럽국가들에 대해서만 맥주와 와인의 데이터를 뽑은 후에 피어슨상관값을 구하는 함수를 적용해야 함)

In [1]:
%matplotlib inline

### [drinks 데이터셋의 기본 정보]

In [14]:
# -*- coding: utf-8 -*-

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

file_path = '../data/drinks.csv'
drinks = pd.read_csv(file_path) # read_csv 함수로 데이터를 Dataframe 형태로 불러옵니다.

In [15]:
print(drinks.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 193 entries, 0 to 192
Data columns (total 6 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   country                       193 non-null    object 
 1   beer_servings                 193 non-null    int64  
 2   spirit_servings               193 non-null    int64  
 3   wine_servings                 193 non-null    int64  
 4   total_litres_of_pure_alcohol  193 non-null    float64
 5   continent                     170 non-null    object 
dtypes: float64(1), int64(3), object(2)
memory usage: 9.2+ KB
None


In [16]:
drinks.head(10)

country  beer_servings  spirit_servings  wine_servings   
0        Afghanistan              0                0              0  \
1            Albania             89              132             54   
2            Algeria             25                0             14   
3            Andorra            245              138            312   
4             Angola            217               57             45   
5  Antigua & Barbuda            102              128             45   
6          Argentina            193               25            221   
7            Armenia             21              179             11   
8          Australia            261               72            212   
9            Austria            279               75            191   

   total_litres_of_pure_alcohol continent  
0                           0.0        AS  
1                           4.9        EU  
2                           0.7        AF  
3                          12.4        EU  
4                           5.9        AF  
5                           4.9       NaN  
6                           8.3        SA  
7                           3.8        EU  
8                          10.4        OC  
9                           9.7        EU

In [5]:
drinks.describe()

beer_servings  spirit_servings  wine_servings   
count     193.000000       193.000000     193.000000  \
mean      106.160622        80.994819      49.450777   
std       101.143103        88.284312      79.697598   
min         0.000000         0.000000       0.000000   
25%        20.000000         4.000000       1.000000   
50%        76.000000        56.000000       8.000000   
75%       188.000000       128.000000      59.000000   
max       376.000000       438.000000     370.000000   

       total_litres_of_pure_alcohol  
count                    193.000000  
mean                       4.717098  
std                        3.773298  
min                        0.000000  
25%                        1.300000  
50%                        4.200000  
75%                        7.200000  
max                       14.400000

### [피처간의 상관관계 탐색]

In [17]:
# 'beer_servings', 'wine_servings' 두 피처간의 상관계수를 계산합니다.
# pearson은 상관계수를 구하는 계산 방법 중 하나를 의미하며, 가장 널리 쓰이는 방법입니다.
corr = drinks[['beer_servings', 'wine_servings']].corr(method = 'pearson')
print(corr)

               beer_servings  wine_servings
beer_servings       1.000000       0.527172
wine_servings       0.527172       1.000000


EU에 속한 데이터만을 가져오기 위해 continent가 EU인 데이터를 가져와
europe에 저장

In [25]:
# 대륙이 EU인 데이터만 가져오기
europe = drinks.loc[drinks['continent']=='EU']
#print(europe['beer_servings'])
#print(europe['wine_servings'])
europe.describe()

beer_servings  spirit_servings  wine_servings   
count      45.000000        45.000000      45.000000  \
mean      193.777778       132.555556     142.222222   
std        99.631569        77.589115      97.421738   
min         0.000000         0.000000       0.000000   
25%       127.000000        81.000000      59.000000   
50%       219.000000       122.000000     128.000000   
75%       270.000000       173.000000     195.000000   
max       361.000000       373.000000     370.000000   

       total_litres_of_pure_alcohol  
count                     45.000000  
mean                       8.617778  
std                        3.358455  
min                        0.000000  
25%                        6.600000  
50%                       10.000000  
75%                       10.900000  
max                       14.400000

europe의 맥주, 와인에 대한 상관계수 구하기

In [26]:
corr = europe[['beer_servings', 'wine_servings']].corr(method = 'pearson')
print(corr)

               beer_servings  wine_servings
beer_servings       1.000000       0.297774
wine_servings       0.297774       1.000000


와인-맥주의 상관계수가 0.297774 이므로 0~1 기준으로 0.3 정도이니 약한 상관계수를 가진다.